src/main/java/de/dennisguse/opentracks/TrackRecordingActivity.java

```diff
@@ -406,4 +405,20 @@ private void onRecordingStatusChanged(TrackRecordingService.RecordingStatus stat
         setLockscreenPolicy();
         setScreenOnPolicy();
     }
-}
+
+    private void onGpsStatusChanged(GpsStatusValue gpsStatusValue) {
+        if (gpsStatusValue.isGpsStarted() && snackbar != null && snackbar.isShown()) {
+            snackbar.dismiss();
+            return;
+        }
+        if (viewBinding.trackRecordingCoordinatorLayout == null || gpsStatusValue != GpsStatusValue.GPS_DISABLED) {
```

Comment:
null check not required, right?
viewbinding could be null (but shouldn't), but all it's child are nonnull :)

Reply:
Yes, you are right and so is AndroidStudio :)